lib/fluent/plugin/out_tdlog.rb

```diff
@@ -78,11 +79,11 @@ def start
       super
 
       client_opts = {
-        :ssl => @use_ssl, :http_proxy => @http_proxy, :user_agent => @user_agent, :endpoint => @endpoint,
+        :ssl => @use_ssl, :http_proxy => @http_proxy, :user_agent => @user_agent,
         :connect_timeout => @connect_timeout, :read_timeout => @read_timeout, :send_timeout => @send_timeout
       }
-      @client = TreasureData::Client.new(@apikey, client_opts)
-
+      @client = TreasureData::Client.new(@apikey, client_opts.merge({:endpoint => @import_endpoint}))
```

Comment:
[nits] 
```suggestion
      @client = TreasureData::Client.new(@apikey, client_opts.merge(endpoint: @import_endpoint))
```

Reply:
It's to represent a hash object explicitly for the coming change of keyword argument handling. See https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/